tests/unit/test_charm.py

```diff
@@ -375,16 +375,16 @@ def test_get_gunicorn_pebble_config(self):
 
     def test_get_gunicorn_pebble_config_error(self):
         """Test the _get_gunicorn_pebble_config function when throwing an error."""
-        expected_output = "ERROR:charm:Error getting pod_env_config: Could not parse Juju config 'environment' as a YAML dict - check \"juju debug-log -l ERROR\""  # noqa: E501
+        expected_error = "Error getting pod_env_config"
         expected_ret = {}
         mock_event = MagicMock()
         with patch('charm.GunicornK8sCharm._make_pod_env') as make_pod_env:
             make_pod_env.return_value = True
 
             with self.assertLogs(level='ERROR') as logger:
-                r = self.harness.charm._get_gunicorn_pebble_config(mock_event)
-                self.assertEqual(r, expected_ret)
-            self.assertEqual(expected_output, logger.output[0])
+                config_output = self.harness.charm._get_gunicorn_pebble_config(mock_event)
```

Comment:
I understand this is out of scope for this PR but keep in mind that we should be testing public methods instead of private ones.